code/obj/machinery/manufacturer.dm

```diff
@@ -26,62 +27,93 @@ TYPEINFO(/obj/machinery/manufacturer)
+	var/wires = 15
 	var/frequency = FREQ_PDA
 	var/net_id = null
 	var/device_tag = "PNET_MANUFACTURER"
+	/// The data terminal attached underfloor to this manufacturer. Allows use of PNET packets
 	var/obj/machinery/power/data_terminal/link = null
-	var/obj/item/card/id/scan = null //! Used when deducting payment for ores from a Rockbox
-
-	// Printing and queues
-	var/original_duration = 0 //! duration of the currently queued print, used to keep track of progress when M.time gets modified weirdly in queueing
-	var/time_left = 0 //! time the current blueprint will take to manufacture
-	var/time_started = 0 //! time the last blueprint was queued
+	/// Card currently scanned into the machine, used when deducting payment for ores from a Rockbox
+	var/obj/item/card/id/scan = null
+
+	/* Printing and queues */
+	/// Original duration of the currently queued print, used to keep track of progress when M.time gets modified weirdly in queueing
+	var/original_duration = 0
+	/// Time left until the current blueprint is complete. Updated on pausing and on starting a new blueprint.
+	var/time_left = 0
+	/// Time the blueprint was queued, or if paused/resumed, the time we resumed the blueprint.
+	var/time_started = 0
+	/// Controls how fast blueprints are produced. Higher speed settings have a exponential effect on power use.
 	var/speed = DEFAULT_SPEED
 	var/repeat = FALSE
-	var/manual_stop = FALSE
-	var/output_cap = 20
+	var/output_cap = MAX_OUTPUT
+	/// A list of manufacture datums in the form of a queue. Blueprints are taken from index 1 and added at the last index
 	var/list/queue = list()
 
-	// Resources/materials
-	var/base_material_class = /obj/item/material_piece //! Base class for material pieces that the manufacturer accepts. Keep this as material pieces only unless you're making larger changes to the system
-	var/free_resource_amt = 0 //! The amount of each free resource that the manufacturer comes preloaded with
-	var/list/obj/item/material_piece/free_resources = list() //! See free_resource_amt; this is the list of resources being populated from
-	var/obj/item/disk/data/floppy/manudrive/manudrive = null
+	/* Resources/materials */
+	/// Base class for material pieces that the manufacturer accepts.
+	/// Keep this as material pieces only unless you're making larger changes to the system,
+	/// Various parts of the code are coupled to assuming that this is a material piece w/ a material.
+	var/base_material_class = /obj/item/material_piece
+	/// The amount of each free resource that the manufacturer comes preloaded with.
+	/// Separate from free_resources() as typically manufacturers use the same amount of each type.
+	var/free_resource_amt = 0
+	/// The types of material pieces of which the manufacturer will be spawned with.
+	/// The amount of each resource is defined on free_resource_amt
+	var/list/obj/item/material_piece/free_resources = list()
```

Comment:
Future work: list of material piece IDs (or however this is done), with keys to determine the material and values to determine the amount.

Reply:
#19150, easy enough so i just went ahead and did it